python/doc/source/development.rst

```diff
@@ -246,6 +246,20 @@ To build a self-contained wheel (include Arrow C++ and Parquet C++), one can set
 Again, if you did not build parquet-cpp, you should omit ``--with-parquet`` and
 if you did not build with plasma, you should omit ``--with-plasma``.
 
+Known issues
+------------
+
+If you're getting some "undefined symbol" errors when importing pyarrow,
+you may have to fix the ABI version used for C++ libraries:
```

Comment:
I think this should be more specific about issues using the conda-forge packages with a newer gcc toolchain. If the user is using a version of gcc, or clang with a base gcc at 5.0 or higher (e.g. with Ubuntu 16.04 or higher), then this flag must be added.